src/udev/udev-connection.h

```diff
@@ -15,7 +15,7 @@ void udev_connection_done(UdevConnection *conn);
 
 static inline int udev_connection_send_ping(UdevConnection *conn) {
         assert(conn);
-        assert(conn->link || conn->uctrl);
+        assert((bool) conn->link ^ (bool) conn->uctrl);
```

Comment:
uh. just do:

`!conn->link != !conn->uctrl`

Reply:
IMHO this requires more mental processing than if explicit cast to bool were used, but if you wish...